File: php/features.php
<?php
declare(strict_types=1);

// UserAgentLookup SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class UserAgentLookupFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new UserAgentLookupBaseFeature();
            case "test":
                return new UserAgentLookupTestFeature();
            default:
                return new UserAgentLookupBaseFeature();
        }
    }
}

File: py/useragentlookup_sdk/features.py
# UserAgentLookup SDK feature factory

from useragentlookup_sdk.feature.base_feature import UserAgentLookupBaseFeature
from useragentlookup_sdk.feature.test_feature import UserAgentLookupTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UserAgentLookupBaseFeature(),
        "test": lambda: UserAgentLookupTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
